action/protocol/staking/protocol.go

```diff
@@ -496,7 +516,11 @@ func (p *Protocol) ActiveCandidates(ctx context.Context, sr protocol.StateReader
 			}
 			list[i].Votes.Add(list[i].Votes, contractVotes)
 		}
-		if list[i].SelfStake.Cmp(p.config.RegistrationConsts.MinSelfStake) >= 0 {
+		active, err := p.isActiveCandidate(ctx, c, list[i])
+		if err != nil {
+			return nil, err
+		}
+		if active {
```

Comment:
1. from the usage here, it is checking current height, not `currentEpochNum+1` height
2. to be safe, i think this check needs to be guarded by hard-fork flag